### PR TITLE
Laboratory work #1, Anastasia Gubarenko - 21FPL2

### PR DESCRIPTION
**Pull request by cybern00dle**

### CODE REVIEW

File: lab_1_keywords_tfidf/main.py [hunk 1]
```diff
@@ -33,7 +41,12 @@ def remove_stop_words(tokens: list[str], stop_words: list[str]) -> Optional[list
 
     In case of corrupt input arguments, None is returned
     """
-    pass
+    for i in tokens[::-1]:
+        if i in stop_words:
+            tokens.remove(i)
+        else:
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:47]: you can just remove `else`

File: lab_1_keywords_tfidf/main.py [hunk 2]
```diff
@@ -33,7 +41,12 @@ def remove_stop_words(tokens: list[str], stop_words: list[str]) -> Optional[list
 
     In case of corrupt input arguments, None is returned
     """
-    pass
+    for i in tokens[::-1]:
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:44]: why do you loop over reverse tokens?

File: lab_1_keywords_tfidf/main.py (hunk 1 above)
Comment (cybern00dle) [reply lab_1_keywords_tfidf/main.py]: done

File: lab_1_keywords_tfidf/main.py (hunk 2 above)
Comment (cybern00dle) [reply lab_1_keywords_tfidf/main.py]: done by appending correct words to an empty list

File: lab_1_keywords_tfidf/main.py [hunk 3]
```diff
@@ -99,7 +163,18 @@ def calculate_tfidf(term_freq: dict[str, float], idf: dict[str, float]) -> Optio
...
-    pass
+    if not check_list_and_dict(term_freq, dict, str, float, False) or not check_list_and_dict(
+            idf, dict, str, float):
+        return None
+    tfidf = {}
+    for i in term_freq.keys():
+        if i in idf.keys():
+            token_idf = idf[i]
+        else:
+            token_idf = log(47 / (0 + 1))
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:174]: why `0 + 1` ?

**Review by artyomtugaryov — APPROVED** [review]

File: lab_1_keywords_tfidf/main.py (hunk 3 above)
Comment (cybern00dle) [reply lab_1_keywords_tfidf/main.py]: done

**Pull request closed without merging**